File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = 0;
        int sell = 1;
        int maxDiff = 0;
        while(sell < prices.size()){
            int currDiff = prices[sell] - prices[buy];
            if(prices[buy] < prices[sell])
                    maxDiff = max(maxDiff,currDiff);
            else{
                buy = sell;
            }
            sell++;
        }
        return maxDiff;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int buyValue = prices[0];
        for(int i = 1; i < prices.length; i++){
            if(prices[i] < buyValue){
                buyValue = prices[i];
            }
            if((prices[i]- buyValue) > profit){
                profit =  prices[i] - buyValue;
            }
        }
        return profit;
    }
}